packages/core/src/mpi/filter-patients/cross-validate-links.ts

```diff
@@ -0,0 +1,49 @@
+import { PatientData } from "../../domain/patient";
+import { isContactMatch } from "./match-contact";
+import { isAddressMatch } from "./match-address";
+
+export function crossValidateInvalidLinks(
+  validLinks: PatientData[],
+  invalidLinks: PatientData[]
+): PatientData[] {
+  if (validLinks.length === 0 || invalidLinks.length === 0) {
+    return [];
+  }
+
+  const crossValidatedLinks: PatientData[] = [];
+
+  for (const invalidLink of invalidLinks) {
+    const isValidated = validateLinkThroughValidLinks(invalidLink, validLinks);
+
+    if (isValidated) {
+      crossValidatedLinks.push(invalidLink);
+    }
+  }
+
+  return crossValidatedLinks;
+}
```

Comment:
I think this can be replaced by a single filter call right?

```
export function crossValidateInvalidLinks(
  validLinks: PatientData[],
  invalidLinks: PatientData[]
): PatientData[] {
  return invalidLinks.filter((invalidLink) => validateLinkThroughValidLinks(invalidLink, validLinks))
}
```

invalidLinks.filter((invalidLink) => validateLinkThroughValidLinks(invalidLink, validLinks))